### ISSUE TITLE
Conv vs Deconv

### ISSUE THREAD

**Issue by taki0112** [taki0112/Cycle_Disco_GAN#2]
# Convolution layer와 Deconvolution layer의 차이
* DCGAN을 보면서 생각난...

## pooling 관점
* Conv에서는 pooling 이라고 하지만, Deconv에서는 Unsampling 이라 함
* Conv에서도 pooling할때 겹치는부분이 있듯이, Deconv에서도 Unpooling할때 겹치는게 생김
* 이걸 Uneven overlap이라고 부름
* 이때 겹치는 숫자들이 평균을 내거나, 합산을 하거나, interpolation하는 방법이 있음 [refer](http://distill.pub/2016/deconv-checkerboard/)
* Unpooling할때 그 위치의 값을 그대로 쓰고, 그 외는 0으로 두는 방법이 있음 (GAN)
* Autoencoder라면, pooling했을때의 값들을 기억하는 메모리를 둠 [refer](http://m.blog.naver.com/sogangori/220455525398)
* Tensorflow에서는 deconv가 아닌, tf.nn.conv2d_transpose가 deconv역할을 하는데, DCGAN 구현코드(김태훈)를 보면, padding 옵션은 딱히 필요 없는듯함

## feature 관점(?)
* Conv는 이미지를 작게작게 분해해서, 세세한 특징들을 알아내고(feature map) 이를 이용함
* Deconv는 세세한 특징을 다시 풀어줘서 새로운 또는 원래의 이미지를 만들어내가는 과정
